nexus/reconfigurator/planning/src/planner.rs

```diff
@@ -1307,24 +1315,27 @@ pub(crate) mod test {
             "1 -> 2 (added additional Nexus zones):\n{}",
             summary.display()
         );
-        assert_eq!(summary.sleds_added.len(), 0);
-        assert_eq!(summary.sleds_removed.len(), 0);
-        assert_eq!(summary.sleds_modified.len(), 1);
-        let changed_sled_id = summary.sleds_modified.first().unwrap();
+        assert_eq!(summary.diff.sleds.added.len(), 0);
+        assert_eq!(summary.diff.sleds.removed.len(), 0);
+        assert_eq!(summary.diff.sleds.modified().count(), 1);
+        let (changed_sled_id, changed_sled) =
+            summary.diff.sleds.modified().next().unwrap();
 
         assert_eq!(*changed_sled_id, sled_id);
         assert_eq!(
-            summary
-                .datasets_on_modified_sled(&sled_id)
-                .unwrap()
-                .datasets
-                .added
-                .len(),
+            changed_sled.diff_pair().datasets_config.datasets.added.len(),
             4
         );
 
-        let zones_added =
-            &summary.zones_on_modified_sled(&sled_id).unwrap().zones.added;
+        let zones_added = &summary
```

Comment:
Possibly the only drawback I see in this PR. However, I also prefer it as all this code is macro generated rather than the handrolled monstrosity that existed before.